edweiss-app/hooks/useListenToMessages.ts

```diff
@@ -0,0 +1,32 @@
+import { useEffect } from 'react';
+
+import { FCMMessage } from '@/model/fcm';
+import messaging from '@react-native-firebase/messaging';
+
+export default function useListenToMessages(handler: (msg: FCMMessage) => void) {
+	useEffect(() => {
+		const unsubscribe = messaging().onMessage(message => {
+			console.log("Foreground message " + JSON.stringify(message));
+
+			if (message.data == undefined)
+				return;
+
+			handler(JSON.parse(message.data.message as any) as any as FCMMessage);
+		});
+		return unsubscribe;
+	}, []);
```

Comment:
The dependency array for the useEffect hook is empty ([]).

Reply:
It does not need to have something in this scenario, since useEffect does as default a single execution once the component/page is loaded